BUG: TestTableSchemaRepr::test_publishes raises AttributeError on mac

### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.


### Reproducible Example

```python
TestTableSchemaRepr::test_publishes - AttributeError: '_GeneratorContextManager' object has no attribute 'args'
```


### Issue Description

I've been seeing this for a couple weeks.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

```
_________________________________________________________ TestTableSchemaRepr.test_publishes __________________________________________________________

self = <pandas.tests.io.formats.test_ipython_compat.TestTableSchemaRepr object at 0x12bb71890>
ip = <IPython.core.interactiveshell.InteractiveShell object at 0x2cc064e10>

    def test_publishes(self, ip):
        ipython = ip.instance(config=ip.config)
        df = DataFrame({"A": [1, 2]})
        objects = [df["A"], df]  # dataframe / series
        expected_keys = [
            {"text/plain", "application/vnd.dataresource+json"},
            {"text/plain", "text/html", "application/vnd.dataresource+json"},
        ]
    
        opt = cf.option_context("display.html.table_schema", True)
        last_obj = None
        for obj, expected in zip(objects, expected_keys):
            last_obj = obj
>           with opt:

pandas/tests/io/formats/test_ipython_compat.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <contextlib._GeneratorContextManager object at 0x2d15ac210>

    def __enter__(self):
        # do not keep args and kwds alive unnecessarily
        # they are only needed for recreation, which is not possible anymore
>       del self.args, self.kwds, self.func
E       AttributeError: '_GeneratorContextManager' object has no attribute 'args'

../../../.pyenv/versions/3.11.7/lib/python3.11/contextlib.py:135: AttributeError
```

--- a/pandas/tests/io/formats/test_ipython_compat.py
+++ b/pandas/tests/io/formats/test_ipython_compat.py
@@ -22,7 +22,8 @@ class TestTableSchemaRepr:
         last_obj = None
         for obj, expected in zip(objects, expected_keys):
             last_obj = obj
-            with opt:
+            with cf.option_context("display.html.table_schema", True):
+                # Can't reuse opt on all systems GH#58055
                 formatted = ipython.display_formatter.format(obj)
             assert set(formatted[0].keys()) == expected
 

